Automattic/vip-go-mu-plugins
# Support multiple Jetpack versions

## Description

Keep multiple jetpack versions inside mu-plugins and allow to pin to one of them with a new define `VIP_JETPACK_PINNED_VERSION`.

Although it supports any value, as long as it matches the directory name, we are going to base it on minor versions, not including patches. The currently available ones are `8.8` and `8.9`. Support for next/beta versions is still work in progress.

You can also skip loading Jetpack altogether by defining `VIP_JETPACK_SKIP_LOAD` to true.

## Reviewing changes

As full Jetpack 8.8.2 and 8.9.1 have been committed, you can check this commit for the actual code changes: https://github.com/Automattic/vip-go-mu-plugins/pull/1749/commits/792bd65a22d052092b62a93f5eb05d5a63f632cf

## Checklist

- n/a This change has relevant unit tests (if applicable).
- (pending) This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Check out PR.
1. Check in the Jetpack dashboard that the active version is 8.9.1
1. Add this define: `define( 'VIP_JETPACK_PINNED_VERSION', '8.8' );`
1. Check in the Jetpack dashboard that the active version is 8.8.2
1. Add this define: `define( 'VIP_JETPACK_SKIP_LOAD', true );`
1. Check that the Jetpack dashboard is not accessible
1. Add this define: `define( 'VIP_JETPACK_PINNED_VERSION', '9.5' );`
1. Check in the Jetpack dashboard that the active version is 8.9.1 (current default version)
